File: 01.JS-Basic/WHILE цикъл/Moving.js
function moving(input) {

    let a = Number(input[0]);
    let b = Number(input[1]);
    let c = Number(input[2]);
    let index = 3;

    let freeSpace = a * b * c;
    let boxes = 0;

    while (freeSpace > boxes) {
        if (input[index] === "Done") {
            console.log(`${freeSpace - boxes} Cubic meters left.`)
            break;
        }
        boxes += Number(input[index]);
        index++;
    }
    if (boxes > freeSpace) {
        console.log(`No more free space! You need ${boxes - freeSpace} Cubic meters more.`)
    }
}
moving(["10",
    "1",
    "2",
    "4",
    "6",
    "Done"])

File: JAVA-Basic/JavaBasic/src/WhileLoop/Moving.java
package WhileLoop;

import java.util.Scanner;

public class Moving {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int w = Integer.parseInt(scan.nextLine());
        int l = Integer.parseInt(scan.nextLine());
        int h = Integer.parseInt(scan.nextLine());

        int freeSpace = w * l * h;

        String input = scan.nextLine();

        while (!input.equals("Done")) {
            int current = Integer.parseInt(input);

            freeSpace -= current;

            if (freeSpace < 0) {
                System.out.printf("No more free space! You need %d Cubic meters more.", Math.abs(freeSpace));
                return;
            }
            input = scan.nextLine();
        }
        System.out.printf("%d Cubic meters left.", freeSpace);
    }
}
